src/lavinmq/http/controller.cr

```diff
@@ -102,11 +102,15 @@ module LavinMQ
         size = 0
         total = 0
         json.array do
-          iterator.each_with_index do |o, i|
-            total += 1
-            next if i < start || start + page_size <= i
-            o.to_json(json)
-            size += 1
+          begin
```

Comment:
Weird that `ameba` doesn't find this but you don't need a begin block, you can rescue in the outer block.